docs/gates.md

```diff
@@ -1,58 +1,241 @@
+axis. `cirq.H` is a π rotation and is equivalent to
+`cirq.HPowGate(exponent=1)`
+
+**S** The square root of Z gate, equivalent to `cirq.Z**0.5`
+
+**T** The fourth root of Z gate, equivalent to `cirq.Z**0.25`.
+
+
+### Two qubit gates
+
+**cirq.CZ / cirq.CZPowGate** The controlled-Z gate.  A two qubit gate that
+phases the |11⟩ state.  `cirq.CZPowGate(exponent=y)` is equivalent to
+`cirq.CZ**t` and has a matrix representation of ``exp(i pi |11⟩⟨11| t)``.
+
+**cirq.CNOT / cirq.CNotPowGate** The controlled-X gate.  This gate swaps the
+|11⟩ and |10⟩ states.  `cirq.CNotPowGate(exponent=t)` is equivalent
+to `cirq.CNOT**t` .
+
+**cirq.SWAP / cirq.SwapPowGate** The swap gate swaps the |01⟩ and |10⟩ states.
+`cirq.SWAP**t` is the same as `cirq.SwapPowGate(exponent = t)`
 
-A class that implements ``Gate`` can be applied to qubits to produce an ``Operation``.
-In order to support functionality beyond that basic task, it is necessary to implement several *magic methods*.
+**cirq.ISWAP / cirq.ISwapPowGate**  The iSwap gate swaps the |01⟩ and |10⟩
+states and adds a relative phase of i.  `cirq.ISWAP**t` is the same as
+`cirq.ISwapPowGate(exponent = t)`
 
-Standard magic methods in python are `__add__`, `__eq__`, and `__len__`.
-Cirq defines several additional magic methods, for functionality such as parameterization, diagramming, and simulation.
-For example, if a gate specifies a `_unitary_` method that returns a matrix for the gate, then simulators will be able to simulate applying the gate.
-Or, if a gate specifies a `__pow__` method that works for an exponent of -1, then `cirq.inverse` will start to work on lists including the gate.
+**Parity gates**: The gates cirq.XX, cirq.YY, and cirq.ZZ are equivalent to
+performing the equivalent one-qubit Pauli gates on both qubits.  The gates
+cirq.XXPowGate, cirq.YYPowGate, and cirq.ZZPowGate are the powers of
+these gates.  
 
-We describe some magic methods below.
 
+### Other Gates
+
+**cirq.MatrixGate**: A gate defined by its unitary matrix in the form of a
+numpy ndarray.
+
+**cirq.WaitGate**:  This gate does nothing for a specified `cirq.Duration`
+amount of time.  This is useful for conducting T1 and T2 decay experiments. 
+
+**cirq.CCNOT, cirq.CCX, cirq.TOFFOLI, cirq.CCXPowGate**: Three qubit gates
+representing the controlled-controlled-X gates.
+
+**cirq.CCZ, cirq.CCZPowGate**: Three qubit gates representing a
+controlled-controlled-Z gate.
+
+**CSWAP, CSwapGate, FREDKIN**: Three qubit gates representing a controlled-SWAP
+gate.
+
+
+## Advanced: Creating your own gates
+
+If the above gates are not sufficient for your use case, it is fairly simple
+to create your own gate.   In order to do so, you can define your class and
+inherit the `cirq.Gate` class and define the functionality in your class.
+
+At minimum, you will need to define either the ``_num_qubits_`` or
+``_qid_shape_`` magic method to define the number of qubits (or qudits) used
```

Comment:
magic method is used before it's defined in the next paragraph. We can either assume it to be a prerequisite or define before using it?

Reply:
Rearranged the two paragraphs so that the definition comes first.